esp/esp/program/setup.py

```diff
@@ -76,24 +73,23 @@ def gen_perm(tup):
 
         if tup[1] is not None:
             new_perm.user=tup[1]
-            new_perm.save()
-            return
         elif tup[1] is None and tup[0].startswith("Student"):
             new_perm.role=Group.objects.get(name="Student")
-            new_perm.save()
-            return
         elif tup[1] is None and tup[0].startswith("Teacher"):
             new_perm.role=Group.objects.get(name="Teacher")
-            new_perm.save()
-            return
+        else:
+            raise ESPError('Invalid permission/deadline: `{}`'.format(tup[1]))
+        new_perm.save()
+        return
 
-        #It's not for a specific user and not a teacher or student deadline
-        for x in ESPUser.getTypes():
-            newnew_perm=Permission(permission_type=new_perm.permission_type, role=Group.objects.get(name=x), start_date=new_perm.start_date, end_date=new_perm.end_date, program=prog)
-            newnew_perm.save()
 
     for perm_tup in perms:
         gen_perm(perm_tup)
+    
+    # Grant onsite permission to admins and onsite user (if an onsite user is available).
+    Permission(permission_type='Onsite', role=Group.objects.get(name="Administrator"), start_date=None, end_date=None, program=prog).save()
```

Comment:
I thought we were going to give admins the "Admin" permission? That should include the "Onsite" permission, but also all of the other permissions.

Reply:
Does it? I thought part of the problem was it didn't, but I will double-check

Reply:
I'm pretty sure the "Administer" permission gives *all* permissions for a program:
https://github.com/learning-unlimited/ESP-Website/blob/a0a1e50fbee39598313869d917f43ffd889cc30b/esp/esp/users/models/__init__.py#L2326